velox/common/memory/MemoryPool.h

```diff
@@ -961,8 +965,14 @@ class MemoryPoolImpl : public MemoryPool {
 
   // Mutex for 'debugAllocRecords_'.
   std::mutex debugAllocMutex_;
+
   // Map from address to 'AllocationRecord'.
   std::unordered_map<uint64_t, AllocationRecord> debugAllocRecords_;
+
+  // Regex for filtering on 'name_' when debug mode is enabled.This allows us to
+  // only track the memory allocations for memory pools whose name matches the
+  // specified regular expression 'debugPoolNameRegex_' if not empty.
+  std::string debugPoolNameRegex_;
```

Comment:
this needs to atomic to support online change through server operation endpoint?

Reply:
Currently not, because this is only set once upon creation and won't change in its lifetime. But if we want to make default root pool's regex_ modifiable in run time then we need to make it thread-safe. And we need to have a lock instead of atomic as std::string is not a primative type.